packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +240,135 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def query_dns_server(fqdn, zone, local_ip)
+      query_dns = Resolv::DNS.new(nameserver: [local_ip], search: [zone], ndots: 1)
+      a_record = query_dns.getresource(fqdn, Resolv::DNS::Resource::IN::A)
+      existing_ip = a_record.address.to_s if a_record
+      soa_record = query_dns.getresource(zone, Resolv::DNS::Resource::IN::SOA)
+      existing_serial = soa_record.serial if soa_record
+      new_serial = (existing_serial + 1).to_s if existing_serial
+      {
+        existing_ip: existing_ip,
+        new_serial: new_serial
+      }
+    end
+
+    def parse_zone_file(zone_file)
+      run_cmd('rndc freeze')
+      zone_file_data = File.read(zone_file).split("\n")
+      # find the character A enclosed by any whitespace chars
+      a_record = zone_file_data.find { |line| !(line =~ /\sA\s/).nil? }
+      existing_ip = a_record.split('A').last.strip if a_record
+      serial_line = zone_file_data.find { |line| line.downcase.include?('serial') }
+      existing_serial = serial_line.split('Serial').first.to_i if serial_line
+      new_serial = (existing_serial + 1).to_s if existing_serial
+      {
+        existing_ip: existing_ip,
+        new_serial: new_serial
+      }
+    end
+
+    def get_dns_data
```

Comment:
rubocop would complain about a method called `get_<whatever>` since it could just be called `whatever`.

But I think the method name should be changed to reflect the fact that it's data representing the future state of the dns records, and not just the current state.

I'm having trouble coming up with a clearer name ...  :thinking:

Reply:
@jturel Renamed to `new_dns_values`